### PR TITLE
Feat/a2 270 ip submission email

### PR DESCRIPTION
**Pull request by topickering**
## Ticket Number

https://pins-ds.atlassian.net/browse/A2-270

## Description of change

Expands IP comment submission:
- add db table for InterestedPartySubmission (ie FO side record)
- add api call to create db entry on submission and trigger notify email if emailaddress provided
- add some submitted checks on pages - if session states comment submitted are returned to enter appeal ref page

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- X My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeals-service-api/src/routes/v2/interested-party-submissions/service.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Create a new interested party
+ *
+ * @param {IPSubmissionData} ipSubmissionData
+ * @returns {Promise<InterestedPartySubmission>}
+ */
+async function createInterestedPartySubmission(ipSubmissionData) {
+	try {
+		return await repo.createInterestedPartySubmission(ipSubmissionData);
+	} catch (error) {
+		logger.error('Error creating Interested Party Submission', { error });
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/routes/v2/interested-party-submissions/service.js:20]: is this backwards? ({error}, 'mesage')

File: packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/add-comments/controller.js [hunk 2]
```diff
@@ -10,7 +11,10 @@ const {
 
 /** @type {import('express').RequestHandler} */
 const addCommentsGet = (req, res) => {
-	if (!confirmInterestedPartySessionAppealReference(req)) {
+	if (
```
Comment (bryn-pins) [inline packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/add-comments/controller.js:14]: feels like these checks should be middleware or shared somehow, so it's handled in one place

File: packages/forms-web-app/src/services/interested-party.service.js [hunk 3]
```diff
@@ -57,12 +52,20 @@ const getInterestedPartyFromSession = (req) => {
...
+};
+
+/**
+ * confirms the interested party from session has caseReference set
  * @param {import('express').Request} req
  * @returns {boolean}
  */
-const confirmInterestedPartySessionAppealReference = (req) => {
-	return !!req.session.interestedParty?.appealNumber;
+const confirmInterestedPartySessionCaseReference = (req) => {
```
Comment (bryn-pins) [inline packages/forms-web-app/src/services/interested-party.service.js:67]: thoughts on renaming these bool checks away from confirm? confirm sounds more like performing an action than checking a bool to me

**Review by bryn-pins — COMMENTED** [review]
some minor thoughts

File: packages/appeals-service-api/src/routes/v2/interested-party-submissions/service.js (hunk 1 above)
Comment (topickering) [reply packages/appeals-service-api/src/routes/v2/interested-party-submissions/service.js:20]: Good spot

File: packages/forms-web-app/src/services/interested-party.service.js (hunk 3 above)
Comment (topickering) [reply packages/forms-web-app/src/services/interested-party.service.js:67]: Will see if I can think of a better name, always the hardest part!

File: packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/enter-address/index.js [hunk 4]
```diff
@@ -5,10 +5,11 @@ const {
 } = require('../../../../validators/interested-parties/enter-address');
 const { validationErrorHandler } = require('../../../../validators/validation-error-handler');
 const asyncHandler = require('@pins/common/src/middleware/async-handler');
+const checkInterestedPartySessionActive = require('../../../../middleware/interested-parties/check-ip-session-set');
 
 const router = express.Router();
 
-router.get('/', asyncHandler(enterAddressGet));
+router.get('/', checkInterestedPartySessionActive, asyncHandler(enterAddressGet));
```
Comment (bryn-pins) [inline packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/enter-address/index.js:12]: should the checks be on post as well?

**Review by bryn-pins — COMMENTED** [review]
need some tests on this please

File: packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/enter-address/index.js (hunk 4 above)
Comment (topickering) [reply packages/forms-web-app/src/routes/file-based-router/comment-planning-appeal/enter-address/index.js:12]: Added

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**